src/supertux/menu/sorted_contrib_menu.cpp

```diff
@@ -38,10 +43,83 @@ SortedContribMenu::SortedContribMenu(std::vector<std::unique_ptr<World>>& worlds
     {
       m_world_folders.push_back(worlds[i]->get_basedir());
       std::string title_str;
+      const auto savegame = Savegame::from_file(worlds[i]->get_savegame_filename());
+
       if (worlds[i]->is_levelset())
-        title_str = "[" + worlds[i]->get_title() + "]";
+      {
+        uint32_t level_count = 0, solved_count = 0;
+        const auto& state = savegame->get_levelset_state(worlds[i]->get_basedir());
+
+        for (const auto& level_state : state.level_states)
+        {
+          if (level_state.filename.empty() || level_state.filename.back() == '~') continue;
+          if (level_state.solved) ++solved_count;
+          ++level_count;
+        }
```

Comment:
I feel like this counting logic doesn't belong here, and could be better reused, if it was moved to a dedicated `LevelsetState` method.